app/controllers/restful_redirect_controller.rb

```diff
@@ -17,8 +24,12 @@ def index
                    end
       redirect_to(:controller => controller, :action => 'show', :id => params[:id])
     else
-      flash_to_session(_("Could not find the given \"%{model}\" record.") % {:model => ui_lookup(:model => params[:model])}, :error)
-      redirect_to(:controller => 'dashboard', :action => 'show')
+      handle_error
     end
   end
+
+  def handle_error
```

Comment:
I seriously thought you can come up with a better name than I, but it's okay :rofl: